.github/workflows/p2p-get-latest-image.yaml

```diff
@@ -0,0 +1,108 @@
+name: p2p-command
+
+on:
+  workflow_call:
+    secrets:
+      env_vars:
+        required: false
+    outputs:
+      version:
+        description: "The latest image in the registry"
+        value: ${{ jobs.get-latest-image.outputs.version }}
+    inputs:
+      registry-path:
+        required: true
+        type: string
+      image-name:
+        required: true
+        type: string
+      source_matrix:
+        required: true
+        type: string
+      dry-run:
+        required: false
+        type: boolean
+        default: false
+      region:
+        required: false
+        type: string
+        default: europe-west2
+      working-directory:
+        required: false
+        type: string
+        default: '.'
+
+
+jobs:
+  lookup:
+    runs-on: ubuntu-latest
+    environment: ${{ fromJson(inputs.source_matrix).include[0]['deploy_env'] }}
```

Comment:
i think its fine to hard code 0 for now

i think this is a reason why the matrix is leaking complexity to the user, its nice that we support multiple envs per stage but also there is no use case for it for the user right now